File: app/src/main/java/com/qimian233/ztool/hook/modules/systemFramework/AllowUntrustedTouch.java
package com.qimian233.ztool.hook.modules.systemFramework;

import com.qimian233.ztool.hook.base.BaseHookModule;

import android.annotation.SuppressLint;

import io.github.libxposed.api.XposedModuleInterface;

import java.lang.reflect.Method;

@SuppressLint("PrivateApi")
public class AllowUntrustedTouch extends BaseHookModule {

    public AllowUntrustedTouch() {}

    @Override
    public String getModuleName() {
        return "allow_untrusted_touch";
    }

    @Override
    public String[] getTargetPackages() {
        return new String[] {"system"};
    }

    @Override
    public void handleLoadPackage(XposedModuleInterface.PackageLoadedParam param) throws Throwable {

    }

    @Override
    public void handleSystemServerStarting(XposedModuleInterface.SystemServerStartingParam param) throws Throwable {
        ClassLoader classLoader = param.getClassLoader();
        Method method = classLoader.loadClass("com.android.server.wm.WindowState")
                .getDeclaredMethod("getTouchOcclusionMode");
        this.xposed.hook(method).intercept(chain -> 2);
    }
}

File: app/src/main/java/com/qimian233/ztool/hook/modules/systemframework/AllowUntrustedTouch.kt
package com.qimian233.ztool.hook.modules.systemframework

import android.annotation.SuppressLint
import com.qimian233.ztool.data.keys.PreferenceKeys
import com.qimian233.ztool.data.keys.ScopeKeys
import com.qimian233.ztool.hook.base.SystemHookModule
import io.github.libxposed.api.XposedModuleInterface.SystemServerStartingParam

@SuppressLint("PrivateApi")
class AllowUntrustedTouch : SystemHookModule() {
    override fun getModuleName(): String = PreferenceKeys.ALLOW_UNTRUSTED_TOUCH.name

    override fun getTargetPackages(): Array<String> = arrayOf(ScopeKeys.SYSTEM_SERVER.packageName)

    override fun handleSystemServerStarting(param: SystemServerStartingParam) {
        val classLoader = param.classLoader
        val method = classLoader.loadClass("com.android.server.wm.WindowState")
            .getDeclaredMethod("getTouchOcclusionMode")
        hookWithId(method, "touch_occlusion_mode") { 2 }
    }
}
